File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String str = "";
        ArrayList<String> res = new ArrayList<>();

        for(int i=0; i<s.length(); i++) {
            if(s.charAt(i) == ' ') {
                if(!str.isEmpty()) {
                    res.add(str);
                    str = "";
                }
            }
            else {
                str += s.charAt(i);
            }
        }
        if(!str.isEmpty()) {
            res.add(str);
        }

        Collections.reverse(res);
        return String.join(" ", res);
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        str = ""
        res = []

        for i in range(len(s)):
            if s[i] == ' ':
                if str:
                    res.append(str)
                    str = ""

            else:
                str += s[i]
        if str:
            res.append(str)
        res.reverse()
        return " ".join(res)